code/modules/atmospherics/machinery/components/unary_devices/machine_connector.dm

```diff
@@ -0,0 +1,111 @@
+///To be used when there is the need of an atmos connection without repathing everything (eg: cryo.dm)
+/obj/machinery/atmospherics/components/unary/machine_connector
+	name = "machine connector"
+	desc = "Internal connector to interface with the pipes."
+
+	can_unwrench = FALSE
+
+	use_power = NO_POWER_USE
+	layer = GAS_FILTER_LAYER
+	hide = TRUE
+	shift_underlay_only = FALSE
+
+	pipe_flags = PIPING_ONE_PER_TURF
+	resistance_flags = INDESTRUCTIBLE
+
+	///Reference to the machine we are connected to
+	var/obj/machinery/connected_machine
+
+/obj/machinery/atmospherics/components/unary/machine_connector/Initialize(mapload, process = TRUE, setdir, init_dir = ALL_CARDINALS, obj/machinery/connecting_machine, direction)
+	dir = direction
+	. = ..()
+	connected_machine = connecting_machine
+	if(!connected_machine)
+		return
+	SSair.start_processing_machine(connected_machine)
+	register_with_machine()
+
+/obj/machinery/atmospherics/components/unary/machine_connector/Destroy()
+	if(connected_machine)
+		connected_machine = null
```

Comment:
```suggestion
	QDEL_NULL(connected_machine)
```

Reply:
the connector must never delete the machine, it must be the other way around

Reply:
right, sorry